crates/database/src/quotes.rs

```diff
@@ -129,6 +132,127 @@ LIMIT 1
         .await
 }
 
+/// Internal structure used for getting Quote with Interactions in one SQL
+/// query.
+#[derive(Clone, sqlx::FromRow)]
+struct QuoteWithInteraction {
+    pub id: QuoteId,
+    pub sell_token: Address,
+    pub buy_token: Address,
+    pub sell_amount: BigDecimal,
+    pub buy_amount: BigDecimal,
+    pub gas_amount: f64,
+    pub gas_price: f64,
+    pub sell_token_price: f64,
+    pub order_kind: OrderKind,
+    pub expiration_timestamp: DateTime<Utc>,
+    pub quote_kind: QuoteKind,
+    pub solver: Address,
+    pub verified: bool,
+    pub index: i32,
+    pub target: Address,
+    pub value: BigDecimal,
+    pub call_data: Vec<u8>,
+}
+
+impl QuoteWithInteraction {
+    fn to_quote(&self) -> Quote {
+        Quote {
+            id: self.id,
+            sell_token: self.sell_token,
+            buy_token: self.buy_token,
+            sell_amount: self.sell_amount.clone(),
+            buy_amount: self.buy_amount.clone(),
+            gas_amount: self.gas_amount,
+            gas_price: self.gas_price,
+            sell_token_price: self.sell_token_price,
+            order_kind: self.order_kind,
+            expiration_timestamp: self.expiration_timestamp,
+            quote_kind: self.quote_kind.clone(),
+            solver: self.solver,
+            verified: self.verified,
+        }
+    }
+
+    fn to_interaction(&self) -> QuoteInteraction {
+        QuoteInteraction {
+            quote_id: self.id,
+            index: self.index,
+            target: self.target,
+            value: self.value.clone(),
+            call_data: self.call_data.clone(),
+        }
+    }
+
+    fn to_quote_with_interactions(items: &[QuoteWithInteraction]) -> Option<QuoteWithInteractions> {
```

Comment:
Bad naming. `to/into` requires `self`.

Reply:
Function moved to `Quote` implementation.